### DISCUSSION TITLE
run a test more than once with pytest.main([])

### DISCUSSION THREAD

**Discussion by jgzzzz** [pytest-dev/pytest#10932] [Q&A] [answered]
I want to  run tests by calling pytest like this: 

```python


tests = ['path\testfile.py::test_init','path\testfile.py::test1','path\testfile.py::test2','path\testfile.py::test_deinit',
'path\testfile.py::test_init','path\testfile.py::test3','path\testfile.py::test4','path\testfile.py::test_deinit']

pytest.main(tests)
``` 
How many tests are defined within init and deinit is up to the tester to define in the testsfile.py(if necesary I can explain this logic), My question is: Pytest only runs  the test_init and test_deinit only once and for the second line on the command both are not runned, so current pytest output on the log is for example like this:

tests collected: 8
- test_init
- test1
- test2
- test_deinit
- test3
- test4

The output I wish to accomplish is:
tests collected: 8
- test_init
- test1
- test2
- test_deinit
- test_init
- test3
- test4
- test_deinit

This tests control some hardware so init and deinit has to be made to prepare test preconditions.
Thanks for any hints

**Comment by jgzzzz** [comment]
searchig for answers I found this post https://stackoverflow.com/questions/46387285/pytest-3-2-2-duplicate-test, but in my case Im not redefining the test_init or test_deinit, I just want to run them again, so the --keep-duplicates has no effect for me.

**Comment by RonnyPfannschmidt** [comment]
That's what fixture scopes and parameterization is for, reinventing them in a call like that isn't going to cut it

**Reply by jgzzzz** [reply]
Ok, I understand, I have tried with fixtures and the code works,  now the thing is I need to also be able to run the same code with my old framework that gives me an error due to the @pytest.fixture on the setup and teardown functions: 

> raise Failed(msg=reason, pytrace=pytrace)

> Failed: Fixture "setup" called directly. Fixtures are not meant to be called directly,

> but are created automatically when test functions request them as parameters.

Is there a way to programatically remove the @pytest.fixture from the setup and teardown?  I have tried the package undecorate but had no sucess.

**Reply by jgzzzz** [reply]
ok sorry the disturbance, the solution was to use test_init.__wrapped and test_deinit.__wrapped and the same fixture from pytest worked with my old framework

**Comment by jgzzzz** [comment, marked as answer]
the pure pytest solution was to add fixtures to the tests on the input arguments, and no longer use/passed init and deinit as tests as on my post above:

```python
@pytest.fixture
def init_system():
    print("This is init_system")


@pytest.fixture
def deinit_system():
    yield
    print("this is deinit_system ")

``` 
 I have explain how I added the fixtureas after test collection on this post: https://stackoverflow.com/questions/76079122/run-a-test-more-than-onces-with-pytest-main/76140534#76140534
